lib/wallet/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -416,10 +418,13 @@ arbitraryChainLength = 10
 -------------------------------------------------------------------------------}
 
 instance Arbitrary TxCBOR where
-    arbitrary = TxCBOR
-        <$> cbor
-        <*> (toEnum <$> choose (0,5))
-      where
+    arbitrary = do
+        c <- cbor
+        n <- elements knownEraIndices
+        case match eraValueSerialize (c,n) of
+            Left _ -> error "impossible era in Arbitrary TxCBOR"
+            Right r -> pure r
```

Comment:
```suggestion
```
The run-time check adds no utility here. If you are worried that this `Arbitrary` instance using `knownEraIndices` might be incorrect, I would add a separate test in the corresponding `QuickCheck` test suite for that.